src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -92,30 +92,29 @@ def filtsmooth(
         KalmanPosterior
             Posterior distribution of the filtered output
         """
-        dataset, times = np.asarray(dataset), np.asarray(times)
         filter_posterior = self.filter(
-            dataset,
-            times,
-            _previous_posterior=_previous_posterior,
+            regression_problem, _previous_posterior=_previous_posterior
         )
         smooth_posterior = self.smooth(filter_posterior)
         return smooth_posterior
 
     def filter(
         self,
-        dataset: np.ndarray,
-        times: np.ndarray,
+        regression_problem: RegressionProblem,
         _previous_posterior: Optional[TimeSeriesPosterior] = None,
     ):
         """Apply Gaussian filtering (no smoothing!) to a data set.
 
         Parameters
         ----------
-        dataset : array_like, shape (N, M)
-            Data set that is filtered.
-        times : array_like, shape (N,)
-            Temporal locations of the data points.
-            The zeroth element in times and dataset is the location of the initial random variable.
+        regression_problem: RegressionProblem
+            A dataclass containing fields for both
+            - locations: array_like, shape (N, )
+            Temporal locations of the (time series) data points, has to be 1-dimensional.
+            The zero-th element is the location of the initial random variable.
+            - observations: array_like, shape(N, M)
+            Observations at the provided locations.
+            The zero-th element is the initial random variable.
```

Comment:
Do you think it would be easier to remove this part of the docstring and instead link to the `RegressionProblem` under `See Also`? I am wondering whether this text here will unknowingly turn into a lie in the near future...